### PR TITLE
D8CORE-3254: basic pages plus

### PR DESCRIPTION
**Pull request by imonroe**
# NOT READY
- Not finished, but please give any early feedback

# Summary
- goes along with https://github.com/SU-SWS/stanford_profile_helper/pull/90

# Needed By (Date)
- When does this need to be merged by?

# Urgency
- How critical is this PR?

# Steps to Test

1. `composer require su-sws/stanford_profile:dev-D8CORE-3254-basicplus --no-update`
2. `composer update`
3. `blt drupal:update` or `blt drupal:install`
4. Then this

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
At first look, these changes make sense to me. once the changes are fixed up in the other 2 repos that i commented on, let me know and i can do a usability review

File: config/sync/ds.field.basic_page_type_content_url.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+id: basic_page_type_content_url
+label: 'Basic Page Type Content URL'
+ui_limit: 'stanford_page|*'
+properties:
+  content:
+    value: "<p>[node:url]</p>\r\n"
+    format: stanford_minimal_html
```
Comment (pookmish) [inline config/sync/ds.field.basic_page_type_content_url.yml:7]: I hate how `plain_text` doesn't work with the tokens. We should fix later that so these ds fields are cleaner.

File: config/sync/field.field.node.stanford_page.su_page_image.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  config:
+    - field.storage.node.su_page_image
+    - media.type.image
+    - node.type.stanford_page
+id: node.stanford_page.su_page_image
+field_name: su_page_image
+entity_type: node
+bundle: stanford_page
+label: 'Page Image'
+description: 'Maximum one image. The <i>"page image"</i> will only appear as a thumbnail image on Teaser and List page display. Use the Banner paragraph to add a Featured Media to a page.'
```
Comment (pookmish) [inline config/sync/field.field.node.stanford_page.su_page_image.yml:14]: Should we also mention it'll be used at the social share image and put some recommendations on size/ratio?

File: config/sync/views.view.basic_page_types.yml [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+  module:
+    - node
+    - stanford_fields
+    - stanford_media
+    - taxonomy
+    - ui_patterns_views
+    - user
+    - views_taxonomy_term_name_depth
+id: basic_page_types
+label: 'Basic Page Types'
```
Comment (pookmish) [inline config/sync/views.view.basic_page_types.yml:19]:
```suggestion
label: 'Basic Pages'
```

File: config/sync/views.view.basic_page_types.yml [hunk 4]
```diff
@@ -0,0 +1,632 @@
...
+    - node.type.stanford_page
+  module:
+    - node
+    - stanford_fields
+    - stanford_media
+    - taxonomy
+    - ui_patterns_views
+    - user
+    - views_taxonomy_term_name_depth
+id: basic_page_types
```
Comment (pookmish) [inline config/sync/views.view.basic_page_types.yml:18]:
```suggestion
id: stanford_basic_pages
```

File: tests/codeception/acceptance/Content/BasicPageCest.php [hunk 5]
```diff
@@ -67,4 +67,40 @@ public function testRevisionPage(AcceptanceTester $I) {
...
+    $title = $faker->text(20);
+    $description = $faker->text(100);
+    $I->logInWithRole('site_manager');
+    $I->amOnPage('/node/add/stanford_page');
+    $I->see('Page Metadata');
+    $I->see('Page Image');
+    $I->see('Basic Page Type');
+    $I->fillField('Title', $title);
+    $I->fillField('Page Description', $description);
+    $I->selectOption('Basic Page Type', 'Research Project');
```
Comment (pookmish) [inline tests/codeception/acceptance/Content/BasicPageCest.php:84]:
```suggestion
    $I->selectOption('Basic Pages', 'Research Project');
```

**Review by pookmish — CHANGES_REQUESTED** [review]

File: config/sync/ds.field.basic_page_type_content_url.yml (hunk 1 above)
Comment (pookmish) [reply config/sync/ds.field.basic_page_type_content_url.yml:7]: this is for a later date. just making note

File: config/sync/core.entity_view_display.node.stanford_page.teaser.yml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+          field_templates: default
+    regions:
+      basic_page_types_category:
+        - su_basic_page_type
+      basic_page_types_title:
+        - node_title
+      basic_page_types_description:
+        - su_page_description
+      basic_page_types_page_image:
+        - su_page_image
```
Comment (pookmish) [inline config/sync/core.entity_view_display.node.stanford_page.teaser.yml:74]: Do these need to update now?

File: config/sync/field.field.paragraph.stanford_lists.su_list_view.yml [hunk 7]
```diff
@@ -20,6 +20,7 @@ default_value_callback: ''
 settings:
   force_default: false
   allowed_views:
+    basic_page_types: basic_page_types
```
Comment (pookmish) [inline config/sync/field.field.paragraph.stanford_lists.su_list_view.yml:23]:
```suggestion
    stanford_basic_pages: stanford_basic_pages
```

File: config/sync/core.entity_view_display.node.stanford_page.teaser.yml (hunk 6 above)
Comment (imonroe) [reply config/sync/core.entity_view_display.node.stanford_page.teaser.yml:74]: actually, I dont think we need this config.  we're not using the teaser display mode for basic pages; we just use the card display mode right?

Comment (imonroe) [reply config/sync/core.entity_view_display.node.stanford_page.teaser.yml:74]: I'm checking now on local.

Comment (imonroe) [reply config/sync/core.entity_view_display.node.stanford_page.teaser.yml]: yeah, it's not necessary.  Removed this config.

**Review by pookmish — APPROVED** [review]

**Pull request merged**